haystack/document_store/milvus2x.py

```diff
@@ -0,0 +1,691 @@
+        """
+        Fetch a document by specifying its text id string
+
+        :param id: ID of the document
+        :param index: Name of the index to get the documents from. If None, the
+                      DocumentStore's default index (self.index) will be used.
+        """
+        documents = self.get_documents_by_id([id], index)
+        document = documents[0] if documents else None
+        return document
+
+    def get_documents_by_id(
+            self, ids: List[str], index: Optional[str] = None, batch_size: int = 10_000
+    ) -> List[Document]:
+        """
+        Fetch multiple documents by specifying their IDs (strings)
+
+        :param ids: List of IDs of the documents
+        :param index: Name of the index to get the documents from. If None, the
+                      DocumentStore's default index (self.index) will be used.
+        :param batch_size: When working with large number of documents, batching can help reduce memory footprint.
+        """
+        index = index or self.index
+        documents = super().get_documents_by_id(ids=ids, index=index)
+        if self.return_embedding:
+            self._populate_embeddings_to_docs(index=index, docs=documents)
+
+        return documents
+
+    def _populate_embeddings_to_docs(self, docs: List[Document], index: Optional[str] = None):
+        index = index or self.index
+        docs_with_vector_ids = []
+        for doc in docs:
+            if doc.meta and doc.meta.get("vector_id") is not None:
+                docs_with_vector_ids.append(doc)
+
+        if len(docs_with_vector_ids) == 0:
+            return
+
+        try:
+            from pymilvus import connections
+            from pymilvus.client.abstract import QueryResult
+        except:
+            raise ImportError("Install pymilvus===2.0.0rc6 version")
+
+        connection = connections.get_connection()
+        connection.load_collection(index)
+
+        ids = [str(doc.meta.get("vector_id")) for doc in docs_with_vector_ids] # type: ignore
+
+        search_result: QueryResult = connection.query(
+            collection_name=index,
+            expr=f'{self.id_field} in [ {",".join(ids)} ]',
+            output_fields=[self.embedding_field],
+        )
+
+        for result, doc in zip(search_result, docs_with_vector_ids):
+            doc.embedding = numpy.array(result["embedding"], dtype="float32")
+
+    def _delete_vector_ids_from_milvus(self, documents: List[Document], index: Optional[str] = None):
+        index = index or self.index
+        existing_vector_ids = []
+        for doc in documents:
+            if "vector_id" in doc.meta:
+                existing_vector_ids.append(str(doc.meta["vector_id"]))
+
+        if len(existing_vector_ids) > 0:
+            expression = f'{self.id_field} in [ {",".join(existing_vector_ids)} ]'
+            res = self.collection.delete(expression)
+            assert len(res) == len(existing_vector_ids)
```

Comment:
If I understand your comment + the CI results here correctly, the delete is still not working here? If yes, we should rather raise an exception here and add a "TODO note"

I saw some discussions in the Milvus slack channel and it seems that the implementation is currently in progress (https://github.com/milvus-io/milvus/issues/7130)